proof-systems/src/darlin/accumulators/dlog.rs

```diff
@@ -0,0 +1,795 @@
+            let label = format!("Test{}", i);
+            labels.push(label.clone());
+            let degree;
+            if segmented {
+                degree = if supported_degree > 0 {
+                    rand::distributions::Uniform::from(1..=supported_degree).sample(rng)
+                } else {
+                    0
+                } * seg_mul;
+            } else {
+                degree = if supported_degree > 0 {
+                    rand::distributions::Uniform::from(1..=supported_degree).sample(rng)
+                } else {
+                    0
+                }
+            }
+            let poly = Polynomial::rand(degree, rng);
+
+            let degree_bound = if let Some(degree_bounds) = &mut degree_bounds {
+                let degree_bound;
+                if segmented {
+                    degree_bound = degree;
+                } else {
+                    let range = rand::distributions::Uniform::from(degree..=supported_degree);
+                    degree_bound = range.sample(rng);
+                }
+                degree_bounds.push(degree_bound);
+                Some(degree_bound)
+            } else {
+                None
+            };
+
+            let hiding_bound = if hiding {
+                if num_points_in_query_set >= degree {
+                    Some(degree)
+                } else {
+                    Some(num_points_in_query_set)
+                }
+            } else {
+                None
+            };
+            println!("Hiding bound: {:?}", hiding_bound);
+
+            polynomials.push(LabeledPolynomial::new(
+                label,
+                poly,
+                degree_bound,
+                hiding_bound,
+            ))
+        }
+        let supported_hiding_bound = polynomials
+            .iter()
+            .map(|p| p.hiding_bound().unwrap_or(0))
+            .max()
+            .unwrap_or(0);
+        println!("supported degree: {:?}", supported_degree);
+        println!("supported hiding bound: {:?}", supported_hiding_bound);
+        println!("num_points_in_query_set: {:?}", num_points_in_query_set);
+        let (ck, vk) = InnerProductArgPC::<G, D>::trim(
+            &pp,
+            supported_degree,
+        )?;
+        println!("Trimmed");
+
+        let (comms, rands) = InnerProductArgPC::<G, D>::commit(&ck, &polynomials, Some(rng))?;
+
+        // Construct "symmetric" query set: every polynomial is evaluated at every
+        // point.
+        let mut query_set = QuerySet::new();
+        let mut values = Evaluations::new();
+        // let mut point = F::one();
+        for _ in 0..num_points_in_query_set {
+            let point = G::ScalarField::rand(rng);
+            for (i, label) in labels.iter().enumerate() {
+                query_set.insert((label.clone(), (format!("{}", i), point)));
+                let value = polynomials[i].evaluate(point);
+                values.insert((label.clone(), point), value);
+            }
+        }
```

Comment:
Why do we sample a "symmetric" query set (i.e. every poly is evaluated at every point)?
Wouldn't it be more representative to, given a point x, randomly choose the polynomials to be evaluated?